### DISCUSSION TITLE
Should I enabled "block all connections not usng VPN" ?

### DISCUSSION THREAD

**Discussion by karam72** [celzero/rethink-app#464] [General]
I use Android 11.

When I open the: 
--> general settings of Android 
--> Connections
--> VPN 
--> It lists on the screen the VPN providers. For now I have only RethinkDNS. 

When I click on the "configure" icon (button which is just right side of the RethinkDNS) it opens me another settings page. It has 2 options:

- Stay connected to VPN
- Block All Connections Not Using VPN

Both of them are disabled. RethinkDNS starts automatically after Android reboot. So what both options means? RethinkDNS already has an built-in functionality which kills the app try to bypass VPN. What are the differences.

**Comment by ignoramous** [comment]
"Stay connected to VPN" must be enabled if you want the VPN to never get killed in the background by the OS, and you want it to start on OS reboot.

"Block all connections not using VPN" must be enabled if you do not want any apps to be able to bypass VPN under any circumstance, including when RethinkDNS itself might crash. That is, with this setting enabled, if RethinkDNS isn't running (ie, has crashed or was stopped), all apps would lose access to the Internet.

I hope this clears it up. If not, let me know.

**Reply by karam72** [reply]
Thank you :) it's clear :+1:

**Reply by karam72** [reply]
@ignoramous  I tested it now. When I enable those options, the excluded apps can not connect to internet.

But if I disable the "Block All Connections Not Using VPN" option the excluded apps can use internet ("Stay connected to VPN" is still enabled).

- Android 11
- RethingDNS 0.53h

**Reply by ignoramous** [reply]
Yes, if you exclude app (to not use the VPN) and *also* enable "Block all connections not using VPN"... then the feature is working as expected. In other words, connections from "excluded apps" are *blocked* because they are *not* using the VPN (as they are _indeed_ excluded from the VPN).

**Reply by karam72** [reply]
Thank you
